src/prover/batch_insertion/mod.rs

```diff
@@ -1,588 +0,0 @@
-/// prover service.
-#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
-pub struct ProverConfiguration {
-    /// The URL at which to contact the semaphore prover service for proof
-    /// generation.
-    pub url: String,
-
-    /// The number of seconds to wait before timing out the transaction.
-    pub timeout_s: u64,
-
-    // TODO Add and query a prover `info` endpoint instead.
-    /// The batch size that the prover is set up to work with. This must match
-    /// the deployed prover.
-    pub batch_size: usize,
-}
-
-/// A representation of the connection to the MTB prover service.
-#[derive(Clone, Debug)]
-pub struct Prover {
-    target_url: Url,
-    client:     reqwest::Client,
-    batch_size: usize,
-    timeout_s:  u64,
-}
-
-impl Prover {
-    /// Constructs a new instance of the Merkle Tree Batcher (or Mtb).
-    ///
-    /// # Arguments
-    /// - `options`: The prover configuration options.
-    pub fn new(options: &ProverConfiguration) -> anyhow::Result<Self> {
-        let target_url = Url::parse(&options.url)?;
-        let timeout_duration = Duration::from_secs(options.timeout_s);
-        let timeout_s = options.timeout_s;
-        let batch_size = options.batch_size;
-        let client = reqwest::Client::builder()
-            .connect_timeout(timeout_duration)
-            .https_only(false)
-            .build()?;
-        let mtb = Self {
-            target_url,
-            client,
-            batch_size,
-            timeout_s,
-        };
-
-        Ok(mtb)
-    }
-
-    /// Creates a new batch insertion prover from the prover taken from the
-    /// database
-    pub fn from_prover_conf(prover_conf: &DbProverConfiguration) -> anyhow::Result<Self> {
-        let target_url = Url::parse(&prover_conf.url)?;
-        let timeout_duration = Duration::from_secs(prover_conf.timeout_s);
-        let client = reqwest::Client::builder()
-            .connect_timeout(timeout_duration)
-            .https_only(false)
-            .build()?;
-
-        Ok(Self {
-            target_url,
-            client,
-            batch_size: prover_conf.batch_size,
-            timeout_s: prover_conf.timeout_s,
-        })
-    }
-
-    pub fn batch_size(&self) -> usize {
-        self.batch_size
-    }
-
-    pub fn timeout_s(&self) -> u64 {
-        self.timeout_s
-    }
-
-    /// Generates a proof term for the provided identity insertions into the
-    /// merkle tree.
-    ///
-    /// # Arguments
-    /// - `start_index`: The index in the merkle tree at which the insertions
-    ///   were started.
-    /// - `pre_root`: The value of the merkle tree's root before identities were
-    ///   inserted.
-    /// - `post_root`: The value of the merkle tree's root after the identities
-    ///   were inserted.
-    /// - `identities`: A list of identity insertions, ordered in the order the
-    ///   identities were inserted into the merkle tree.
-    pub async fn generate_proof(
```

Comment:
Do we need this functionality for tests? It looks like its not used anywhere in the core codebase.